bundles/org.eclipse.passage.lic.jetty/src/org/eclipse/passage/lic/internal/jetty/JettyHandler.java

```diff
@@ -48,73 +45,36 @@ public JettyHandler(Function<NetRequest, NetResponse> handler) {
-		response.setStatus(200);
-		long contentLength = -1;
-		for (HttpField field : request.getHeaders()) {
-			HttpHeader header = field.getHeader();
-			if (header == null) {
-				continue;
-			}
-			switch (header) {
-			case CONTENT_LENGTH:
-				response.getHeaders().add(field);
-				contentLength = field.getLongValue();
-				break;
-			case CONTENT_TYPE:
-				response.getHeaders().add(field);
-				break;
-			case TRAILER:
-				response.setTrailersSupplier(HttpFields.build());
-				break;
-			case TRANSFER_ENCODING:
-				contentLength = Long.MAX_VALUE;
-				break;
-			}
-		}
-		if (contentLength > 0) {
-			//FIXME: AF: here we should somehow consider our "handler" function
-			Content.copy(request, response, Response.newTrailersChunkProcessor(response), callback);
-		} else {
-			callback.succeeded();
-		}
+	public boolean handle(Request request, Response answer, Callback callback) throws Exception {
+		NetResponse response = handler.apply(new JettyRequest(request));
+		write(response, request, answer, callback);
 		return true;
 
 	}
 
-//	@Override
-//	public void handle(String target, Request request, HttpServletRequest wrapper, HttpServletResponse envelope)
-//			throws IOException, ServletException {
-//		write(response(wrapper), envelope);
-//		request.setHandled(true);
-//	}
-
-	private NetResponse response(Request request) {
-		return handler.apply(new JettyRequest(request));
-	}
-
-	private void write(NetResponse response, HttpServletResponse envelope) throws IOException {
-		envelope.setContentType(response.contentType().contentType());
-		envelope.setCharacterEncoding("UTF-8"); //$NON-NLS-1$
+	private void write(NetResponse response, Request request, Response answer, Callback callback) throws IOException {
 		if (response.failed()) {
-			envelope.sendError(response.error().code(), response.error().message());
+			Error error = response.error();
+			Response.writeError(request, answer, callback, error.code(), error.message());
+			return;
+		}
+		answer.setStatus(HttpStatus.OK_200);
+		if (!response.carriesPayload()) {
 			return;
 		}
-		envelope.setStatus(HttpServletResponse.SC_OK);
-		if (response.carriesPayload()) {
-			byte[] payload;
-			try {
-				payload = response.payload();
-			} catch (Exception e) {
-				throw new IOException(e);
-			}
-			envelope.setContentLength(payload.length);
-			try (PrintWriter out = envelope.getWriter()) {
-				out.write(new String(payload, Charset.forName("UTF-8"))); //$NON-NLS-1$
-				out.flush();
-			}
-			envelope.flushBuffer();
+		byte[] payload;
+		try {
+			payload = response.payload();
+		} catch (Exception e) {
+			throw new IOException(e);
 		}
+		answer.getHeaders().put(HttpHeader.CONTENT_LENGTH, payload.length);
+		answer.getHeaders().put(HttpHeader.CONTENT_TYPE, type(response) + "; charset=UTF-8"); //$NON-NLS-1$
+		answer.write(true, ByteBuffer.wrap(payload), callback);
+	}
+
+	private String type(NetResponse netResponse) {
```

Comment:
It's perfectly normal for a method name to be composite. 
But for anything that carries data (vars, params, fields) - it's not. 
This parameter has compound name.

Reply:
Ah ok. Then I misunderstood your previous comment.
I restored the method name as `contentType()` and used a single noun name for the parameter.